rfcs/0140-simple-package-paths.md

```diff
@@ -0,0 +1,234 @@
+---
+feature: simple-package-paths
+start-date: 2022-09-02
+author: Silvan Mosberger
+co-authors: Nixpkgs Architecture Team
+shepherd-team: (names, to be nominated and accepted by RFC steering committee)
+shepherd-leader: (name to be appointed by RFC steering committee)
+related-issues: https://github.com/NixOS/nixpkgs/pull/211832
+---
+
+# Summary
+[summary]: #summary
+
+Auto-generate trivial top-level attribute definitions in `pkgs/top-level/all-packages.nix` from a directory structure that matches the attribute name.
+This makes it much easier to contribute new packages packages, since there's no more guessing needed as to where the package should go, both in the ad-hoc directory categories and in `pkgs/top-level/all-packages.nix`.
+
+# Motivation
+[motivation]: #motivation
+
+- (Especially new) package contributors are having a hard time figuring out which files to add or edit. These are very common questions:
```

Comment:
Playing devil's advocate: Contributing *new* packages is the last thing that should be made easier for first time contributors, since they are less likely to stick around and actually maintain their additions.

Reply:
```suggestion
- Contributors of new packages, especially Nix and Nixpkgs beginners, are having a hard time figuring out which files to add or edit. These are very common questions:
```

Reply:
As a rude elitist, I note that addressing some usability issues but not the others allows us to influence what type of contributions will happen as a part of the learning process and which contributions usually come from experienced collaborators.

On the other hand, there is «you need to spend effort learning this, then it starts making sense», and there is «you will learn that this is not actually consistent so you can just guess». I believe here we have the latter case…

```suggestion
- (Especially new) package contributors are having a hard time figuring out which files to add or edit. These are very common questions:
+It is not obvious to package contributors where to add files. Even experienced contributors don't make the choices in the same way. These are very common questions:
```

Reply:
I don't think we need to be very nit-picky with the sentencing here. Just to avoid the ambiguity I'll change it to:

> It is not obvious to package contributors where to add files or which ones to edit. These are very common questions:

@sternenseemann This change doesn't address your original comment though, but with the discussion here, do you consider that addressed?